src/applications/PopulationAtlases.cpp

```diff
@@ -134,13 +41,13 @@ std::vector<typename ImageType::Pointer> PopulationAtlases::GeneratePopualtionAt
 		atlasimage->Allocate();
 		atlasimage->FillBuffer(0);
 
-		for (int j = 0; j < labels.size(); j++)
+		for (int j = 0; j < atlas_labels.size(); j++)
 		{
-			if (labels[j] == atlas_no)
+			if (atlas_labels[j] == atlas_no)
 			{
 				counter++;
 				std::cout << "Subject # = " << counter << std::endl;
-				ImageType::Pointer currentImagePointer = ReadNiftiImage<ImageType>(inputdirectory + "/" + subjects[j]);
+				ImageType::Pointer currentImagePointer = cbica::ReadImage<ImageType>(image_paths[j]);
```

Comment:
Please add some comments about what is happening within the loops.

Reply:
added comments